### ISSUE TITLE
different results than tape itself

### ISSUE THREAD

**Issue by ljharb** [Raynos/collapsed-assert#3]
I've started a branch to try out using this package [here](https://github.com/ljharb/es-abstract/compare/collapsed_assert).

However, the tests fail when they should pass: https://travis-ci.com/github/ljharb/es-abstract/builds/176444693

Oddly enough, they pass in node < 8. Any ideas?

**Comment by Raynos** [comment]
Maybe i did not implement `throws` properly ? it proxies to node core throws()

**Comment by ljharb** [comment]
it doesn’t call into tape’s “throws”? altho they should be the same I’m sure there’s differences.

**Comment by Raynos** [comment]
it uses `assert.throws` to check whether it should report a failure to `tape.throws` ;

**Comment by Raynos** [comment]
This avoids a bunch of `ok foobar` lines in TAP output by not calling into `tape` for assertions that passed.

**Comment by ljharb** [comment]
that seems a bit more complex than `var threw = false; try { maybeThrows(); } catch (e) { threw = true; }`; would that work?

**Comment by Raynos** [comment]
Yeah but not with the regex parameter.

**Comment by ljharb** [comment]
Although #4 fixes it for throws/doesNotThrow, I'm seeing a similar issue with `t.equal`, since node's `assert.equal` changed from non-strict by default to strict by default at some point. Also tape 4 uses deep-equal v1 which defaults to loose, and tape 5 uses deep-equal v2 which defaults to strict.

It seems like it'd be better to use tape's API rather than node's assert. is that something you're interested in?

**Comment by Raynos** [comment]
It would be good to use tape's API but I dont know what public or private API there available to use.

We want to run the assertions but not report them unless they fail in which case we report all of them.

I'd rather not add a direct dependency on `tape` though as currently the library has zero dependencies.

**Comment by ljharb** [comment]
Without a direct dependency, I'm not sure how it could be injected in a way that would work for tape, and tap, and "insert tap-producer here". If it's meant to be tied to tape's API, then it should have a dependency on tape.

**Comment by Raynos** [comment]
We take `t` as an argument, the current design is meant to mirror `assert` in node core. 

If assert made a breaking change i didnt pay attention to then this module should make the same breaking change :D

**Comment by ljharb** [comment]
It's made a number.

If it's meant to mirror assert, then that's unfortunately much less useful, since assert throws and doesn't produce TAP.

**Comment by Raynos** [comment]
I didnt mean it like that, it should implement assert like methods like tap/tape do. 

We can update the `equal` method to loose or not loose deep equal, either is fine.

**Comment by ljharb** [comment]
Right, but the way tap and tape do that varies by version, hence the need for a peer dep.
